File: BussinesLogic/Servicios/ServicioTipoIndicador.cs
using DataAccess.StoredProcedures;
using Dominio.Dto;
using Dominio.Entiti;
using Dominio.Interfaces.Aplicacion.TipoIndicador;
using Dominio.Interfaces.Infraestructura.BaseDatos;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading.Tasks;

namespace BussinesLogic.Servicios
{
    public class ServicioTipoIndicador : IServicioTipoIndicador
    {
        public readonly IRepositorioTipoIndicador repositorioTipoIndicador;

        public ServicioTipoIndicador(IRepositorioTipoIndicador repositorioTipoIndicador)
        {
            this.repositorioTipoIndicador = repositorioTipoIndicador;
        }
        public async Task<DtoRespuestaSP> InsertarTipoIndicador(EntitiTipoIndicador entitiTipoIndicador)
        {
            return await this.repositorioTipoIndicador.InsertarTipoIndicador(entitiTipoIndicador);
        }
        public async Task<DtoRespuestaSP> ActualizarTipoIndicador(EntitiTipoIndicador entitiTipoIndicador)
        {
            return await this.repositorioTipoIndicador.ActualizarTipoIndicador(entitiTipoIndicador);
        }
        public async Task<DtoRespuestaSP> ActualizarValorIndicador(EntitiValorIndicador entitiValorIndicador)
        {
            return await this.repositorioTipoIndicador.ActualizarValorIndicador(entitiValorIndicador);
        }
        public async Task<DtoRespuestaSP> EliminarTipoIndicador(int idTipoIndicador)
        {
            return await this.repositorioTipoIndicador.EliminarTipoIndicador(idTipoIndicador);
        }
        public async Task<DtoTipoIndicador> ObtenerTipoIndicadorPorID(int idTipoIndicador)
        {
            return await this.repositorioTipoIndicador.ObtenerTipoIndicadorPorID(idTipoIndicador);
        }
        public async Task<List<DtoValorIndicador>> ObtenerValoresIndicadorPorID(int idTipoIndicador)
        {
            return await this.repositorioTipoIndicador.ObtenerValoresIndicadorPorID(idTipoIndicador);
        }

        public async Task<List<DtoTipoIndicador>> ObtenerTipoIndicador()
        {
            return await this.repositorioTipoIndicador.ObtenerTiposIndicadores();
        }

        public async Task<List<DtoTipoIndicador>> ObtenerTipoIndicadorActivos()
        {
            return await this.repositorioTipoIndicador.ObtenerTiposIndicadoresActivos();
        }
    }
}

File: CX/ClientApp/src/servicios/ServicioTipoIndicador.js
import { ProcesarDatosApi } from "./ApiFetch";
const controlador = "tipoindicador";

export const AgregarTipoIndicador = async (data) => {
    const url = `${controlador}/InsertarTipoIndicador`;
    return await ProcesarDatosApi('POST', url, data);
}

export const ActualizarTipoIndicador = async (data) => {
    const url = `${controlador}/ActualizarTipoIndicador`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const ActualizarValorIndicador = async (data) => {
    const url = `${controlador}/ActualizarValorIndicador`;
    return await ProcesarDatosApi('PUT', url, data);
}

export const InactivarTipoIndicador = async (id) => {
    const url = `${controlador}/EliminarTipoIndicador?idTipoIndicador=${id}`;
    return await ProcesarDatosApi('DELETE', url);
}

export const ObtenerTipoIndicadorPorId = async (id) => {
    const url = `${controlador}/ObtenerTipoIndicadorPorId/${id}`;
    return await ProcesarDatosApi('GET', url);
}

export const ObtenerTiposIndicadores = async () => {
    const url = `${controlador}/ObtenerTiposIndicadores`; 
    return await ProcesarDatosApi('GET', url);
}
export const ObtenerValoresIndicadorPorID = async (id) => {
    const url = `${controlador}/ObtenerValoresIndicadorPorID/${id}`;
    return await ProcesarDatosApi('GET', url);
}
export const ObtenerTiposIndicadoresActivos = async () => {
    const url = `${controlador}/ObtenerTiposIndicadoresActivos`;
    return await ProcesarDatosApi('GET', url);
}


